src/pages/Admin/Admin.tsx

```diff
@@ -284,15 +309,57 @@ const EventsComponent: React.FC<VenueDetailsPartProps> = ({ venue }) => {
   const [showCreateEventModal, setShowCreateEventModal] = useState(false);
   const [showDeleteEventModal, setShowDeleteEventModal] = useState(false);
   const [editedEvent, setEditedEvent] = useState<WithId<VenueEvent>>();
-  //filter out events that have already finished
-  const filteredEvents = events?.filter(
-    (ev) =>
-      (ev.start_utc_seconds + ev.duration_minutes * 60) * 1000 > Date.now()
+  const [filterPastEvents, setFilterPastEvents] = useState(false);
+  const [filterText, setFilterText] = useState("");
+
+  const upcomingEvents = useMemo(
+    () =>
+      filterPastEvents
+        ? events?.filter(
+            (ev) =>
+              (ev.start_utc_seconds + ev.duration_minutes * 60) * 1000 >
+              Date.now()
+          )
+        : events,
+    [events, filterPastEvents]
+  );
+
+  const fuse = useMemo(
+    () =>
+      upcomingEvents
+        ? new Fuse(upcomingEvents, { keys: ["name", "description", "host"] })
+        : upcomingEvents,
```

Comment:
does it makes sense for const fuse = useMemo(()=>upcomingEvents ? ... : undefined..